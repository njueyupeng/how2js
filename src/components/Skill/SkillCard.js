import React from 'react'
import { Card, Progress } from 'antd'
import PropTypes from 'prop-types'
import 'antd/lib/card/style/index.css'
import 'antd/lib/progress/style/index.css'

const SkillCard = (props) => {
  const { item, percent } = props

  return (
    <React.Fragment>
      <div>
        <Card
          title={item}
          bordered={false}
          size="small"
          style={{ width: 165, textAlign: 'center', paddingRight: '10px', paddingLeft: '10px' }}
        >
          <Progress
            type="circle"
            strokeColor={{
              '0%': '#1890ff',
              '25%': '#1F75C5',
              '50%': '#25598A',
              '75%': '#27537C',
              '100%': '#2c3e50',
            }}
            percent={`${percent}`}
            format={() => ''}
            width={100}
            style={{ justifyContent: 'center' }}
          />
        </Card>
      </div>
    </React.Fragment>
  )
}

SkillCard.propTypes = {
  item: PropTypes.string.isRequired,
  percent: PropTypes.string.isRequired,
}

export default SkillCard
