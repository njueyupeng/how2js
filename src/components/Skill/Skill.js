import React from 'react'
import SkillCard from './SkillCard'

const Skill = () => {
  return (
    <React.Fragment>
      <div className="skill">
        <SkillCard item="Testing" percent="90" />
        <SkillCard item="Agile" percent="90" />
        <SkillCard item="JavaScript" percent="85" />
        <SkillCard item="Java" percent="80" />
        <SkillCard item="Regex" percent="75" />
        <SkillCard item="C#" percent="70" />
      </div>

      {/* --- STYLES --- */}
      <style jsx>{`
        .skill {
          display: flex;
          flex-direction: row;
          flex-wrap: wrap;
          justify-content: center;
          border-radius: 10px;
          padding-bottom: 20px;
        }
      `}</style>
    </React.Fragment>
  )
}

export default Skill
