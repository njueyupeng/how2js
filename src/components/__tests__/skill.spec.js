import React from 'react'
import { create } from 'react-test-renderer'
import Skill from '../Skill'

describe('Skill Component', () => {
  it('renders correctly', () => {
    const tree = create(<Skill />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
