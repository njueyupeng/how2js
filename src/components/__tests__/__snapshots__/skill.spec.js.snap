// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill Component renders correctly 1`] = `
Array [
  <div
    className="skill"
  >
    <div>
      <div
        className="ant-card ant-card-small"
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "textAlign": "center",
            "width": 165,
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Testing
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <div
              className="ant-progress-inner ant-progress-circle-gradient"
              style={
                Object {
                  "fontSize": 21,
                  "height": 100,
                  "width": 100,
                }
              }
            >
              <svg
                className="ant-progress-circle"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <defs>
                  <linearGradient
                    id="ant-progress-gradient-1"
                    x1="100%"
                    x2="0%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#1890ff"
                    />
                    <stop
                      offset="25%"
                      stopColor="#1F75C5"
                    />
                    <stop
                      offset="50%"
                      stopColor="#25598A"
                    />
                    <stop
                      offset="75%"
                      stopColor="#27537C"
                    />
                    <stop
                      offset="100%"
                      stopColor="#2c3e50"
                    />
                  </linearGradient>
                </defs>
                <path
                  className="ant-progress-circle-trail"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  stroke={null}
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": null,
                      "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
                <path
                  className="ant-progress-circle-path"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  opacity={1}
                  stroke="url(#ant-progress-gradient-1)"
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": Object {
                        "0%": "#1890ff",
                        "100%": "#2c3e50",
                        "25%": "#1F75C5",
                        "50%": "#25598A",
                        "75%": "#27537C",
                      },
                      "strokeDasharray": "265.77873849369655px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
              </svg>
              <span
                className="ant-progress-text"
                title=""
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ant-card ant-card-small"
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "textAlign": "center",
            "width": 165,
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Agile
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <div
              className="ant-progress-inner ant-progress-circle-gradient"
              style={
                Object {
                  "fontSize": 21,
                  "height": 100,
                  "width": 100,
                }
              }
            >
              <svg
                className="ant-progress-circle"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <defs>
                  <linearGradient
                    id="ant-progress-gradient-2"
                    x1="100%"
                    x2="0%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#1890ff"
                    />
                    <stop
                      offset="25%"
                      stopColor="#1F75C5"
                    />
                    <stop
                      offset="50%"
                      stopColor="#25598A"
                    />
                    <stop
                      offset="75%"
                      stopColor="#27537C"
                    />
                    <stop
                      offset="100%"
                      stopColor="#2c3e50"
                    />
                  </linearGradient>
                </defs>
                <path
                  className="ant-progress-circle-trail"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  stroke={null}
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": null,
                      "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
                <path
                  className="ant-progress-circle-path"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  opacity={1}
                  stroke="url(#ant-progress-gradient-2)"
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": Object {
                        "0%": "#1890ff",
                        "100%": "#2c3e50",
                        "25%": "#1F75C5",
                        "50%": "#25598A",
                        "75%": "#27537C",
                      },
                      "strokeDasharray": "265.77873849369655px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
              </svg>
              <span
                className="ant-progress-text"
                title=""
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ant-card ant-card-small"
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "textAlign": "center",
            "width": 165,
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              JavaScript
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <div
              className="ant-progress-inner ant-progress-circle-gradient"
              style={
                Object {
                  "fontSize": 21,
                  "height": 100,
                  "width": 100,
                }
              }
            >
              <svg
                className="ant-progress-circle"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <defs>
                  <linearGradient
                    id="ant-progress-gradient-3"
                    x1="100%"
                    x2="0%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#1890ff"
                    />
                    <stop
                      offset="25%"
                      stopColor="#1F75C5"
                    />
                    <stop
                      offset="50%"
                      stopColor="#25598A"
                    />
                    <stop
                      offset="75%"
                      stopColor="#27537C"
                    />
                    <stop
                      offset="100%"
                      stopColor="#2c3e50"
                    />
                  </linearGradient>
                </defs>
                <path
                  className="ant-progress-circle-trail"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  stroke={null}
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": null,
                      "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
                <path
                  className="ant-progress-circle-path"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  opacity={1}
                  stroke="url(#ant-progress-gradient-3)"
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": Object {
                        "0%": "#1890ff",
                        "100%": "#2c3e50",
                        "25%": "#1F75C5",
                        "50%": "#25598A",
                        "75%": "#27537C",
                      },
                      "strokeDasharray": "251.01325302182448px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
              </svg>
              <span
                className="ant-progress-text"
                title=""
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ant-card ant-card-small"
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "textAlign": "center",
            "width": 165,
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Java
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <div
              className="ant-progress-inner ant-progress-circle-gradient"
              style={
                Object {
                  "fontSize": 21,
                  "height": 100,
                  "width": 100,
                }
              }
            >
              <svg
                className="ant-progress-circle"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <defs>
                  <linearGradient
                    id="ant-progress-gradient-4"
                    x1="100%"
                    x2="0%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#1890ff"
                    />
                    <stop
                      offset="25%"
                      stopColor="#1F75C5"
                    />
                    <stop
                      offset="50%"
                      stopColor="#25598A"
                    />
                    <stop
                      offset="75%"
                      stopColor="#27537C"
                    />
                    <stop
                      offset="100%"
                      stopColor="#2c3e50"
                    />
                  </linearGradient>
                </defs>
                <path
                  className="ant-progress-circle-trail"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  stroke={null}
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": null,
                      "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
                <path
                  className="ant-progress-circle-path"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  opacity={1}
                  stroke="url(#ant-progress-gradient-4)"
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": Object {
                        "0%": "#1890ff",
                        "100%": "#2c3e50",
                        "25%": "#1F75C5",
                        "50%": "#25598A",
                        "75%": "#27537C",
                      },
                      "strokeDasharray": "236.24776754995247px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
              </svg>
              <span
                className="ant-progress-text"
                title=""
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ant-card ant-card-small"
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "textAlign": "center",
            "width": 165,
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Regex
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <div
              className="ant-progress-inner ant-progress-circle-gradient"
              style={
                Object {
                  "fontSize": 21,
                  "height": 100,
                  "width": 100,
                }
              }
            >
              <svg
                className="ant-progress-circle"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <defs>
                  <linearGradient
                    id="ant-progress-gradient-5"
                    x1="100%"
                    x2="0%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#1890ff"
                    />
                    <stop
                      offset="25%"
                      stopColor="#1F75C5"
                    />
                    <stop
                      offset="50%"
                      stopColor="#25598A"
                    />
                    <stop
                      offset="75%"
                      stopColor="#27537C"
                    />
                    <stop
                      offset="100%"
                      stopColor="#2c3e50"
                    />
                  </linearGradient>
                </defs>
                <path
                  className="ant-progress-circle-trail"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  stroke={null}
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": null,
                      "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
                <path
                  className="ant-progress-circle-path"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  opacity={1}
                  stroke="url(#ant-progress-gradient-5)"
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": Object {
                        "0%": "#1890ff",
                        "100%": "#2c3e50",
                        "25%": "#1F75C5",
                        "50%": "#25598A",
                        "75%": "#27537C",
                      },
                      "strokeDasharray": "221.48228207808043px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
              </svg>
              <span
                className="ant-progress-text"
                title=""
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ant-card ant-card-small"
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "textAlign": "center",
            "width": 165,
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              C#
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <div
              className="ant-progress-inner ant-progress-circle-gradient"
              style={
                Object {
                  "fontSize": 21,
                  "height": 100,
                  "width": 100,
                }
              }
            >
              <svg
                className="ant-progress-circle"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <defs>
                  <linearGradient
                    id="ant-progress-gradient-6"
                    x1="100%"
                    x2="0%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      offset="0%"
                      stopColor="#1890ff"
                    />
                    <stop
                      offset="25%"
                      stopColor="#1F75C5"
                    />
                    <stop
                      offset="50%"
                      stopColor="#25598A"
                    />
                    <stop
                      offset="75%"
                      stopColor="#27537C"
                    />
                    <stop
                      offset="100%"
                      stopColor="#2c3e50"
                    />
                  </linearGradient>
                </defs>
                <path
                  className="ant-progress-circle-trail"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  stroke={null}
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": null,
                      "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
                <path
                  className="ant-progress-circle-path"
                  d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                  fillOpacity="0"
                  opacity={1}
                  stroke="url(#ant-progress-gradient-6)"
                  strokeLinecap="round"
                  strokeWidth={6}
                  style={
                    Object {
                      "stroke": Object {
                        "0%": "#1890ff",
                        "100%": "#2c3e50",
                        "25%": "#1F75C5",
                        "50%": "#25598A",
                        "75%": "#27537C",
                      },
                      "strokeDasharray": "206.7167966062084px 295.3097094374406px",
                      "strokeDashoffset": "-0px",
                      "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                    }
                  }
                />
              </svg>
              <span
                className="ant-progress-text"
                title=""
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <style
    jsx={true}
  >
    
        .skill {
          display: flex;
          flex-direction: row;
          flex-wrap: wrap;
          justify-content: center;
          border-radius: 10px;
          padding-bottom: 20px;
        }
      
  </style>,
]
`;
